### ISSUE TITLE
detectSum: is a strange name for an unconditional loop.

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#8215]
```
detectSum: aBlock
	"Evaluate aBlock with each of the receiver's elements as the argument. 
	Return the sum of the answers."
	| sum |
	sum := 0.
	self do: [:each | 
		sum := (aBlock value: each) + sum].  
	^ sum
```

**Comment by gcorriga** [comment]
I suggest replacing it with `#accumulate:` or `#accumulateWith:`.
If people are ok with this name I can submit a PR.

Should `#detectSum:` be kept as deprecated for now?

**Comment by svenvc** [comment]
This looks to be the same as #sum: at least in Pharo 7

**Comment by Ducasse** [comment]
I think that I like accumulate: and accumulate:sumBlock: :)
Now there was a discussion on sum and vincent proposed a nice solution that we integrate (I do not remember it but it should be in bug tracker - but github search is more than bad). I remember that Sven was also participating to the discussion at that time) the point was the fact that this method forced the domain to implement +, while a block would be much better. So I 
In any case we would have to deprecate it.

**Comment by Ducasse** [comment]
@sven indeed we could already simply deprecate detectSum: to sum: it would be already an improvement.

**Comment by gcorriga** [comment]
#2225 is related to this, but I found it quite hard to follow the discussion.

**Comment by Ducasse** [comment]
Thanks Yes it was long. Now I remember that vincent proposed something nice. We should ask him. @VincentBlondeau

**Comment by VincentBlondeau** [comment]
@Ducasse here it is:
https://github.com/pharo-project/pharo/issues/6373

merged in https://github.com/pharo-project/pharo/pull/6453
`
detectSum:` is optimized for number computation and strictly equivalent to `sum:` in result.

I suggest to deprecate `detectSum:`
